src/test/unit/algorithms/TemporalMemoryTest.cpp

```diff
@@ -0,0 +1,943 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2013-2015, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ----------------------------------------------------------------------
+ */
+
+/** @file
+ * Implementation of unit tests for TemporalMemory
+ */
+
+#include <cstring>
+#include <fstream>
+#include <stdio.h>
+#include <nupic/math/StlIo.hpp>
+#include <nupic/types/Types.hpp>
+#include <nupic/utils/Log.hpp>
+
+#include "TemporalMemoryTest.hpp"
+
+using namespace std;
+
+namespace nupic {
+
+  void TemporalMemoryTest::setup(TemporalMemory& tm, UInt numColumns)
+  {
+    vector<UInt> columnDim;
+    columnDim.push_back(numColumns);
+    tm.initialize(columnDim);
+  }
+
+  void TemporalMemoryTest::RunTests()
+  {
+    setup(tm, 2048);
+
+    testInitInvalidParams();
+    testActivateCorrectlyPredictiveCells();
+    testActivateCorrectlyPredictiveCellsEmpty();
+    testBurstColumns();
+    testBurstColumnsEmpty();
+    testLearnOnSegments();
+    testComputePredictiveCells();
+    testBestMatchingCell();
+    testBestMatchingCellFewestSegments();
+    testBestMatchingSegment();
+    testLeastUsedCell();
+    testAdaptSegment();
+    testAdaptSegmentToMax();
+    testAdaptSegmentToMin();
+    testPickCellsToLearnOn();
+    testPickCellsToLearnOnAvoidDuplicates();
+    testColumnForCell1D();
+    testColumnForCell2D();
+    testColumnForCellInvalidCell();
+    testCellsForColumn1D();
+    testCellsForColumn2D();
+    testCellsForColumnInvalidColumn();
+    testNumberOfColumns();
+    testNumberOfCells();
+    testMapCellsToColumns();
+    testWrite();
+
+    // Depreciated serialization method (incomplete due to lack of load & save 
+    // in Connections and Random classes). Replaced by Cap'n Proto read & write.
+    //testSaveLoad();
```

Comment:
testSaveLoad can be enabled (commented out lines in load and save), but requires Connections and Random to support the old serialization method. Could someone create separate PRs for those two changes?